### PR TITLE
#20369  fixes running the import content-type command

### LINKED ISSUE #20369: We need to create a bash script that imports Content types in our native json format
The customer should be able to pulldown a dotCMS git repo with a folder containing the definition of CT and import them from a dotCMS instance via the CLT Which internally relies on the _**/api/v1/contenttype**_  endpoint to do the import/export 

The CLT will use internally CURL But it will initially be written in bash. or python. 
The idea is to be able to checkout a directory from say github and then push that directory into a dotCMS instance.

Define a folder structure that we can also use when generating bundles that can be used to push these assets into a dotCMS instance.

Best efforts for knowable IDs:  We will need to update the creation of sites, content types, folders and files and use a sha256 of their hostName+language+path as the identifier for these assets.  If that hash already exists in the db, we will just create a new UUID and use it instead.
See: #19443

Pass a directory  and push its content into a dotCMS instance.
1. Sites 
   - file extension pattern = `.site.json`
   - add site based on siteName
   - include identifier in the json
   
2. ContentTypes (should be the same format as we get from the API)
   - file extension pattern = `.contenttype.json`
   - do a find on the passed in directory for
   - if content type exists (by var) then we update
   - else we put a new content type
   
3. File Assets - via webdav (can reuse the webdav endpoint)
   - https://dotcms.com/docs/latest/scripting-webdav
   - Best efforts for knowable IDs:  We will need to update the creation of folders and files and use a sha256 of their hostName+language+path as the identifier for these assets.  If that hash already exists in the db, we will just create a new UUID and use it instead.
   - Use the bundle layout found here: https://docs.google.com/spreadsheets/d/1iO7lq23Rdcev_-sFB5WdX8-KrEsy96dUTdMhvHY8IrE/edit#gid=0

The scripts/files should accept ContentTypes/fields with and without inodes.  If the contenttype definition includes the inode  and they do not match the ones we have in the dotCMS instance, the script should fail.

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
Ok, this thing fixes an issue when running the content-type import command like 
`./dot-cli --content-type ../cli/bundles/bike-shop/content-types/bike.contenttype.json`

The JSON file by default comes with a site-identifier place holder that gets replaced when running the **--bundle** command
When running the script with the **--content-type** modifier on the file was getting sent with an empty site-identifier string.

When the Content-type resource finds an empty site-id the Content-Type gets re-built all the fields leaving them empty. 
I'm not changing that server-side logic but instead, I am adding validation on the script that prevents sending empty identifiers.

Also, there was a problem removing files used for different purposes by script. Those were replaced by true temp files.

### CODE REVIEW

**Review by nollymar — COMMENTED** [review]
I would think that a validation should be included in the endpoint. It doesn't make any sense to create a corrupted content type (via Rest) if the site is empty

**Review by jdotcms — APPROVED** [review]

**Comment by fabrizzio-dotCMS** [comment]
> I would think that a validation should be included in the endpoint. It doesn't make any sense to create a corrupted content type (via Rest) if the site is empty

Yeah, a that's a pretty valid observation..  I'll add it

File: dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java [hunk 1]
```diff
@@ -420,15 +420,17 @@ public ContentType save(ContentType contentType, List<Field> newFields, List<Fie
     }
     // Sets the host:
     try {
-      if (contentType.host() == null || contentType.fixed()) {
-        final List<Field> existinFields = contentType.fields();
+      if (UtilMethods.isNotSet(contentType.host()) || contentType.fixed()) {
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java:423]: any test?

Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java:423]: done.

**Review by freddyucv — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/contenttype/test/ContentTypeAPIImplTest.java [hunk 2]
```diff
@@ -2255,4 +2259,145 @@ public void testIsContentTypeAllowedReturnsFalse() throws Exception {
...
+			//Here we test that the fields are still present even when the site is null
+			Assert.assertFalse(savedContentType.fields().isEmpty());
+
+			assertNotNull(savedContentType.host());
+
+			final String expected = "default".equals(testCase.expected) ? APILocator.getHostAPI()
+					.findDefaultHost(APILocator.systemUser(), false).getIdentifier()
+					: testCase.expected;
+
+			assertEquals(savedContentType.host(), expected);
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotcms/contenttype/test/ContentTypeAPIImplTest.java:2349]: small observation: the expected results is always the first param :)

**Review by nollymar — APPROVED** [review]

**Pull request merged**